lib/puppet/type/cisco_interface_portchannel.rb

```diff
@@ -0,0 +1,131 @@
+# Manages a Cisco PortChannel Interface.
+#
+# Dec 2015
+#
+# Copyright (c) 2013-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+Puppet::Type.newtype(:cisco_interface_portchannel) do
+  @doc = %q(
+    "Manages a Cisco PortChannel Interface.
+     Any resource dependency should be run before the interface resource.
```

Comment:
IMO, we don't need this resource dependency statement.  Users managing the domain should understand the dependencies.